packages/create-react-admin/src/app.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { useReducer } from 'react';
+import { Box, Text } from 'ink';
+import {
+    InitialProjectConfiguration,
+    ProjectConfiguration,
+} from './ProjectState.js';
+import { generateProject } from './generateProject.js';
+import { StepDataProvider } from './StepDataProvider.js';
+import { StepAuthProvider } from './StepAuthProvider.js';
+import { StepResources } from './StepResources.js';
+
+type Props = {
+    name: string | undefined;
+};
+
+const stepReducer = (
+    state: ProjectConfiguration,
+    action
+): ProjectConfiguration => {
+    switch (state.step) {
+        case 'data-provider':
+            return {
+                ...state,
+                step: 'auth-provider',
+                dataProvider: action.value,
+                resources:
+                    action.value === 'ra-data-fakerest'
+                        ? ['posts', 'comments']
+                        : state.resources,
+            };
+        case 'auth-provider':
+            return {
+                ...state,
+                step:
+                    state.dataProvider === 'ra-data-fakerest'
+                        ? 'generate'
+                        : 'resources',
+                authProvider: action.value,
+            };
+        case 'resources':
+            return {
+                ...state,
+                step: 'generate',
+                resources: action.value,
+            };
+        case 'generate':
+            return {
+                ...state,
+                step: 'finish',
+            };
+        default:
+            return state;
+    }
+};
+
+export default function App({ name = 'my-admin' }: Props) {
+    const [state, dispatch] = useReducer(stepReducer, {
+        ...InitialProjectConfiguration,
+        name,
+    });
+    const handleSubmit = (value: any) => {
+        dispatch({ value });
+    };
+
+    if (state.step === 'data-provider') {
+        return <StepDataProvider onSubmit={handleSubmit} />;
+    }
+    if (state.step === 'auth-provider') {
+        return <StepAuthProvider onSubmit={handleSubmit} />;
+    }
+    if (state.step === 'resources') {
+        return <StepResources onSubmit={handleSubmit} />;
+    }
+    if (state.step === 'generate') {
+        generateProject(state).then(() => {
+            dispatch({});
+        });
+        return <Text>Generating your application...</Text>;
+    }
+    return (
+        <>
+            <Text>
+                Generated your application{' '}
+                <Text color="blue">{state.name}</Text> with{' '}
+                <Text color="blue">{state.dataProvider}</Text>{' '}
+                {state.authProvider !== 'none' ? (
+                    <>
+                        and <Text color="blue">{state.authProvider}</Text>
```

Comment:
For the local auth provider, the CLI must explain how to login (it's not clear that the users are in a json file)

Reply:
Explain it where and when ? We don't have a system that would allow a package to add custom instructions. Let's do that in a future version.

Reply:
Explain it in the terminal once the install is finished. Something like

> Use "johndoe"/"password" to login (or any of the username/password from the users.json file)

If the current package system doesn't allow to do it, we should change it. For instance, why not add a special key in each package's `package.json` (which, by the way, is more like a `config.json`)?

Reply:
Isn't explaining it in the README enough?

Reply:
No, the expected behavior is to have enough info in the console to be able to run the app right away

Reply:
Done